query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/ExecutionContextTest.java

```diff
@@ -308,4 +311,133 @@ public void testSelectionsLinkedHashSet() {
     assertEquals(
         Expression.newBuilder().setFunction(minFunction).build(), selectionsIterator.next());
   }
+
+  @Test
+  public void testSetTimeSeriesPeriod() {
+
+    // no group by
+    Builder builder = QueryRequest.newBuilder();
+    QueryRequest queryRequest = builder.build();
+    ExecutionContext context = new ExecutionContext("test", queryRequest);
+    assertEquals(Optional.empty(), context.getTimeSeriesPeriod());
+
+    // group by on column
+    builder.addGroupBy(
+        Expression.newBuilder()
+            .setColumnIdentifier(
+                ColumnIdentifier.newBuilder().setColumnName("Trace.service_name")));
+    queryRequest = builder.build();
+    context = new ExecutionContext("test", queryRequest);
+    assertEquals(Optional.empty(), context.getTimeSeriesPeriod());
+
+    // group by on other functions
+    Function count =
+        Function.newBuilder()
+            .setFunctionName("Count")
+            .setAlias("myCountAlias")
+            .addArguments(
+                Expression.newBuilder()
+                    .setColumnIdentifier(ColumnIdentifier.newBuilder().setColumnName("Trace.id")))
+            .build();
+    builder.addGroupBy(Expression.newBuilder().setFunction(count));
+    queryRequest = builder.build();
+    context = new ExecutionContext("test", queryRequest);
+    assertEquals(Optional.empty(), context.getTimeSeriesPeriod());
+  }
+
+  @Test
+  public void testSetTimeSeriesPeriodForTimeSeriesRequest() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addGroupBy(
+        Expression.newBuilder()
+            .setFunction(createTimeColumnGroupByFunction("SERVICE.startTime", 15))
```

Comment:
An important part of this test is the unit. Rather than having `createTimeColumnGroupByFunction` take in 15 and assume it's seconds, pass it "15:SECONDS" (and also test a different unit).